tools/clang/unittests/HLSLExec/ShaderOpTest.h

```diff
@@ -218,6 +218,18 @@ class ShaderOp {
     }
     return result;
   }
+  LPCSTR GetShaderArguments(ShaderOpShader *pShader) {
+    if (!pShader || !pShader->Arguments) return nullptr;
+    LPCSTR result = pShader->Arguments;
+    if (result[0] == '@') {
+      for (auto && S : Shaders) {
```

Comment:
May I ask why auto && instead of auto &?

Reply:
I suspect this is just how it was copied from `GetShaderText` which uses this, but doesn't really need to. It's harmless anyway.

Reply:
Correct, this is copied from `GetShaderText` and this syntax is used in multiple places in the `ShaderOpTest` code. It means  universal reference that will bind to anything (lvalue or rvalue).